packages/standard-minifier-js/plugin/minify-js.js

```diff
@@ -1,105 +1,254 @@
 import { extractModuleSizesTree } from "./stats.js";
+import fs from 'fs';
+
+export function getConfig() {
+  try{
+    const meteorAppDir = process.cwd();
+    // read the meteor project pakcgae.json file
+    const packageJson = fs.readFileSync(`${meteorAppDir}/package.json`, 'utf8');
+    const meteorConfig = JSON.parse(packageJson).meteor;
+    return meteorConfig;
+  } catch (error) {
+    console.log(error);
+    return {};
+  }
+};
```

Comment:
It would be nice if the meteor-tool provided the modern config to build plugins instead of each one having to check itself. Meteor already does this for other global options, such as if lazy compilation is available or if HMR is enabled.

Reply:
Could you expand on that?

Like a `Meteor.config`(with all that package.json meteor property has)?

Reply:
We already have a [MeteorConfig](https://github.com/meteor/meteor/blob/2892718148a4453f31223b96e09aa7814ad80fae/tools/project-context.js#L1770) class in tools/project-context.js. In [`_readProjectMetadata`](https://github.com/meteor/meteor/blob/2892718148a4453f31223b96e09aa7814ad80fae/tools/project-context.js#L492-L494), add:

``` js
self.meteorConfig = new MeteorConfig({
  appDirectory: self.projectDir,
});
global.meteorConfig = self.meteorConfig;
```

After you run `global.meteorConfig._ensureInitialized()`, `global.meteorConfig._config` holds the parsed JSON at compile time without re-initializing. We can also add new methods for getting directly the modern config in the MeteorConfig class.

I’ll need to update the related code on my end as well.

Reply:
I submitted a PR that refactors areas using the Meteor config: https://github.com/meteor/meteor/pull/13743

Once it’s merged into the 3.3 branch, we can spread it here and adjust as needed.